src/particles/PhysicsParticles.hpp

```diff
@@ -323,6 +424,116 @@ template <typename ContainerType, ParticleType particleType> class PhysicsPartic
 		}
 	}
 
+	// Implementation of CIC particle creation
+	void createCICParticles(amrex::MultiFab &state, int lev, amrex::Real current_time, amrex::Real dt) const override
+	{
+		if (container_ != nullptr) {
+			const int mass_idx = this->getMassIndex();
+			if (mass_idx >= 0 && mass_idx + 3 < ContainerType::ParticleType::NReal) {
+				const double particle_creation_time_1 = this->getParam1();
+				const double particle_creation_time_2 = this->getParam2();
+
+				for (amrex::MFIter mfi = container_->MakeMFIter(lev); mfi.isValid(); ++mfi) {
+					// TODO(cch): I don't know whether this is needed. WarpX has this.
+					amrex::Gpu::synchronize();
```

Comment:
This isn't necessary. It's done in WarpX in order to compute the time it takes to process the particles in each FAB.